framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -753,40 +754,58 @@ public AnnotationMirror getAnnotationFromJavaExpressionString(
             String expression, Tree tree, TreePath path, Class<? extends Annotation> clazz)
             throws FlowExpressionParseException {
 
-        FlowExpressions.Receiver expressionObj =
-                getReceiverFromJavaExpressionString(expression, path);
-        return getAnnotationFromReceiver(expressionObj, tree, clazz);
+        Receiver expressionObj = getReceiverFromJavaExpressionString(expression, path);
+        Set<AnnotationMirror> annotations = getAnnotationsFromReceiver(expressionObj, tree);
+        return AnnotationUtils.getAnnotationByClass(annotations, clazz);
     }
+
     /**
-     * Returns the primary annotation on a receiver.
+     * Produces the annotation mirror that represents the given expression String in this type
+     * factory's type system.
      *
-     * @param receiver the receiver for which the annotation is returned
-     * @param tree current tree
-     * @param clazz the Class of the annotation
-     * @return the annotation on expression or null if one does not exist
+     * @param expression a Java expression
+     * @param tree the tree at the location to parse the expression
+     * @param currentPath location at which expression is evaluated
+     * @throws FlowExpressionParseException thrown if the expression cannot be parsed
+     * @return an AnnotationMirror representing the type in the store at the given location from
+     *     this type factory's type system, or null if one is not available
      */
-    public AnnotationMirror getAnnotationFromReceiver(
-            FlowExpressions.Receiver receiver, Tree tree, Class<? extends Annotation> clazz) {
+    public AnnotationMirror getAnnotationFromJavaExpressionString(
+            String expression, Tree tree, TreePath currentPath)
+            throws FlowExpressionParseException {
+        Receiver rec = getReceiverFromJavaExpressionString(expression, currentPath);
+        Set<AnnotationMirror> annotations = getAnnotationsFromReceiver(rec, tree);
+        if (annotations.size() == 1) {
+            return annotations.iterator().next();
+        } else {
+            return null;
+        }
+    }
 
-        AnnotationMirror annotationMirror = null;
+    /**
+     * Returns the set of annotations on a receiver.
+     *
+     * @param receiver the receiver whose annotation is returned
+     * @param tree current tree
+     * @return a set containing the annotations on the receiver, or the empty set if none are found
+     */
+    public Set<AnnotationMirror> getAnnotationsFromReceiver(Receiver receiver, Tree tree) {
         if (CFAbstractStore.canInsertReceiver(receiver)) {
```

Comment:
```
import java.util.Arrays;
import org.checkerframework.common.value.qual.MinLen;

public class PlumeFail {
    void method() {
        @SuppressWarnings({"index", "value"})
        String @MinLen(1)[] args = getArray();
        String[] argArray = Arrays.copyOfRange(args, 1, args.length);
    }
    String[] getArray() {
        return null;
    }
}
```
The above code is failing on in plume lib.
```
PlumeFail.java:9: error: [argument.type.incompatible] incompatible types in argument.
        String[] argArray = Arrays.copyOfRange(args, 1, args.length);
                                                     ^
  found   : @UpperBoundUnknown int
  required: @LTEqLengthOf("args") int
1 error
```
Before these changes, if the exact annotation requested didn't exist, then the value from the factory was used.  I tried switching this code so that the factory annotation was always used for fields and local variables.  But that caused a crash.  (I don't see a quick solution to fix the crash.)  We can work around the crash by casting:
```
        String[] args = (String @MinLen(1)[]) getArray();
```
That seems ok to me, but I don't know if this will cause a lot of new errors in the case studies. You can revert my last commit and your changes to this code.  (Though you will need to add `CFAbstractStore.canInsertReceiver(receiver)` to your original code.)